jwst/resample/resample_spec.py

```diff
@@ -228,6 +236,9 @@ def do_drizzle(self, xmin=0, xmax=0, ymin=0, ymax=0, **pars):
             output_model.meta.wcs.bounding_box = bb
             output_model.meta.filename = obs_product
 
+            if self.drizpars['blendheaders']:
+                self.blend_output_metadata(output_model)
+
```

Comment:
I notice that the implementation in `resample` has a couple extra lines here to save the value of the DATAMODL (meta.model_type) keyword from the original output model and then store it again after it gets clobbered by the blending process (i.e. the output model_type gets overwritten by the model_type from the inputs during blending). This may not be necessary anymore, now that we're storing resampled data in the same kind of model as the input data, but it would be worth checking to confirm that we no longer need to worry about that. A few regtests should tell us the answer.

Reply:
Do you want to include that in the scope of this PR, or check in on it in a different one?

Reply:
I guess we can just merge this as-is and watch for any fallout from the regtests. A patch could be added later, if necessary.

Reply:
No worries, I'll check it out now and see what happens. Shouldn't take long in any case

Reply:
@hbushouse So, I commented out the lines that you mentioned in `resample` and then ran the `test_nircam_image` regression tests. Looks like they passed, and I was able to confirm that the test cases included model blending.

Reply:
OK, good. I would also try running a couple spectroscopic regtests on this branch to make sure we're also good for spectroscopic resampled data models (e.g. SlitModel, MultiSlitModel).

Reply:
@hbushouse In running the regression tests under `test_nirspec_fs_spec3` the differences seem to make sense (I think). For example, in the primary header, some date and time related keywords are different, and other differences are a result of the number of extensions being different (due to blending)

Reply:
That sounds as expected. Some of the date/time keywords get modified by blending to correspond to the min/max/sum of all inputs. So I think we're good!

Reply:
Just tried it for `nirspec_mos_spec3`, and there are the same differences as expected, although it appears for one of the parametrized sources there are differences in the `crf` file. Any thoughts for that one?

Reply:
Differences in pixel values?

Reply:
In the DQ images it looks like

Reply:
Did you happen to rebase against master, which includes the PR from @stscieisenhamer on DQ handling in the `outlier_detection` step? If not, I've also occasionally seen small numbers of differences in `crf` files just due to differences in numerical precision when running the test on my own machine vs jwcalibdev, such that some pixel values are just slightly different enough to cross from one side of the detection threshold to the other, resulting in more or fewer DQ flags.